services/ui-src/src/components/banners/AdminBannerProvider.tsx

```diff
@@ -1,66 +1,120 @@
-import { useState, createContext, ReactNode, useMemo, useEffect } from "react";
-// types
-import { AdminBannerData, AdminBannerShape } from "types/banners";
+import { createContext, ReactNode, useMemo, useEffect } from "react";
 // utils
-import { useStore, deleteBanner, getBanner, writeBanner } from "utils";
-// verbiage
-import { bannerErrors } from "verbiage/errors";
+import { AdminBannerData, AdminBannerShape } from "types/banners";
 import { bannerId } from "../../constants";
+import { bannerErrors } from "verbiage/errors";
+// api
+import {
+  checkDateRangeStatus,
+  deleteBanner,
+  getBanner,
+  useStore,
+  writeBanner,
+} from "utils";
 
 const ADMIN_BANNER_ID = bannerId;
 
 export const AdminBannerContext = createContext<AdminBannerShape>({
   fetchAdminBanner: Function,
   writeAdminBanner: Function,
   deleteAdminBanner: Function,
-  isLoading: false as boolean,
-  errorMessage: undefined,
 });
 
 export const AdminBannerProvider = ({ children }: Props) => {
-  const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [error, setError] = useState<string>();
-
   // state management
-  const { bannerData, setAdminBanner, clearAdminBanner } = useStore();
+  const {
+    bannerData,
+    setBannerData,
+    isBannerActive,
+    setIsBannerActive,
+    isBannerLoading,
+    setIsBannerLoading,
+    bannerErrorMessage,
+    setBannerErrorMessage,
+    isBannerDeleting,
+    setIsBannerDeleting,
+  } = useStore();
 
   const fetchAdminBanner = async () => {
-    setIsLoading(true);
+    setIsBannerLoading(true);
     try {
       const currentBanner = await getBanner(ADMIN_BANNER_ID);
       const newBannerData = currentBanner?.Item || {};
-      setAdminBanner(newBannerData);
+      setBannerData(newBannerData);
+      setBannerErrorMessage("");
     } catch (e: any) {
-      setIsLoading(false);
-      setError(bannerErrors.GET_BANNER_FAILED);
+      // 404 expected when no current banner exists
+      if (!e.toString().includes("404")) {
+        setBannerErrorMessage(bannerErrors.GET_BANNER_FAILED);
+      }
     }
-    setIsLoading(false);
+    setIsBannerLoading(false);
   };
 
   const deleteAdminBanner = async () => {
-    await deleteBanner(ADMIN_BANNER_ID);
-    clearAdminBanner();
+    setIsBannerDeleting(true);
+    try {
+      await deleteBanner(ADMIN_BANNER_ID);
+      await fetchAdminBanner();
+    } catch (error: any) {
+      setBannerErrorMessage(bannerErrors.DELETE_BANNER_FAILED);
+    }
+    setIsBannerDeleting(false);
   };
 
   const writeAdminBanner = async (newBannerData: AdminBannerData) => {
-    await writeBanner(newBannerData);
-    setAdminBanner(newBannerData);
+    try {
+      await writeBanner(newBannerData);
+    } catch (e: any) {
+      setBannerErrorMessage(bannerErrors.CREATE_BANNER_FAILED);
+    }
+    await fetchAdminBanner();
```

Comment:
The same does not apply here, since the data saved by the API may not be identical to the data sent by the UI, due to validation on the API side. If I understand correctly.